src/ui/wxWidgets/AddEditPropSheetDlg.cpp

```diff
@@ -1347,52 +1343,95 @@ static struct {short pv; wxString name;}
+  // an interval but no date, which is interpreted as "Never".  We are going to ignore
+  // it and use the user-set default.  If an expiry change is made, the entry will
+  // be re-written correctly.
+  int defInterval = PWSprefs::GetInstance()->GetPref(PWSprefs::DefaultExpiryDays);
 
-  wxCommandEvent dummy;
   wxDateTime exp;
   if (m_tttExpirationTime == 0) { // never expires
     m_DatesTimesExpireOnCtrl->SetValue(false);
     m_DatesTimesExpireInCtrl->SetValue(false);
     m_DatesTimesNeverExpireCtrl->SetValue(true);
-    exp = wxDateTime::Now();
-    dummy.SetEventObject(m_DatesTimesNeverExpireCtrl);
+    m_OriginalRecurring = false;
+    m_OriginalButton = m_DatesTimesNeverExpireCtrl;
+    m_Item.SetXTimeInt(0);  // Special case: No date, there should be no interval
+    m_ExpirationTimeInterval = defInterval;
+    exp = TodayPlusInterval();
+
   } else {
-    exp = wxDateTime(m_tttExpirationTime);
+    exp = wxDateTime(m_tttExpirationTime).GetDateOnly();  // Remove time part
+    m_OriginalDayttt = exp.GetTicks();
+
     if (m_ExpirationTimeInterval == 0) { // expiration specified as date
       m_DatesTimesExpireOnCtrl->SetValue(true);
       m_DatesTimesExpireInCtrl->SetValue(false);
       m_DatesTimesNeverExpireCtrl->SetValue(false);
       m_DatesTimesExpiryTimeCtrl->Enable(false);
-      m_Recurring = false;
-      dummy.SetEventObject(m_DatesTimesExpireOnCtrl);
-    } else { // exp. specified as interval
+      m_OriginalRecurring = false;
+      m_OriginalButton = m_DatesTimesExpireOnCtrl;
+      m_ExpirationTimeInterval = defInterval;
+
+    } else { // exp. specified as recurring interval
       m_DatesTimesExpireOnCtrl->SetValue(false);
       m_DatesTimesExpireInCtrl->SetValue(true);
       m_DatesTimesNeverExpireCtrl->SetValue(false);
       m_DatesTimesExpiryDateCtrl->Enable(false);
       m_DatesTimesExpiryTimeCtrl->SetValue(m_ExpirationTimeInterval);
-      m_Recurring = true;
-      dummy.SetEventObject(m_DatesTimesExpireInCtrl);
+      m_OriginalRecurring = true;
+      m_OriginalButton = m_DatesTimesExpireInCtrl;
+      exp = TodayPlusInterval();
+    }
+  }
+  m_DatesTimesRecurringExpiryCtrl->Enable(m_OriginalRecurring);
+
+  // The date picker controls on different platforms (i.e. Mac vs. GTK)
+  // behave differently with respect to handling the time portion.  This
+  // results in different values when converting to or from time_t and
+  // false or missed change detections.
+  // GTK seems to remove the time part, macOS preserves it.
+  // Since we only care about the date for expiration, let's just
+  // remove the time wherever we need the date.
+  // Note the wxWidgets documentation says Today() returns the
+  // time part set to 0, and Today() and Now() both use the local time zone.
+  m_DatesTimesExpiryDateCtrl->SetValue(exp);
+  m_tttExpirationTime = exp.GetTicks();
+  m_Recurring = true;
+
+  if (m_OriginalDayttt) {
+    wxString rstr;
+    int interval = IntervalFromDate(wxDateTime(m_OriginalDayttt));
```

Comment:
```suggestion
    int interval = IntervalFromDate(exp);
```
or is it crucial that we include the time part here?

Reply:
This is building a string to represent the number of days until expiration, so it needs the original date.  At this point, exp (expiryDate) may have changed.